### PR TITLE
SD CI kernel support, and some misc cleanup

### PR DESCRIPTION
**Pull request by msheiny**
The main goal of this PR was to build an SD kernel config with XEN support (used strictly for CI). During the course of that work, came across a couple of other places for improvement:

* Module swap -- performed a module swap with `command` -> `uri`
* support for append build config - I didnt want to maintain a parallel config for CI, felt it would be better to maintain just a small diff. so added a feature to allow providing an over-ride var file with config changes.
* support for using env vars in build cmd -- fairly straight forward. the make environment supports passing env vars.

### CODE REVIEW

File: tasks/fetch_grsecurity_files.yml [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
+    force_basic_auth: yes
+    user: "{{ grsecurity_build_download_username }}"
+    password: "{{ grsecurity_build_download_password }}"
+    dest: "{{ item.dest }}"
+    creates: "{{ item.dest }}"
+  with_items:
+    - url: "{{ grsecurity_patch_url }}"
+      dest: "{{ grsecurity_build_download_directory }}/{{ grsecurity_patch_filename }}"
+    - url: "{{ grsecurity_signature_url  }}"
+      dest: "{{ grsecurity_build_download_directory }}/{{ grsecurity_signature_filename }}"
```
Comment (conorsch) [inline tasks/fetch_grsecurity_files.yml]: Fantastic, happy to ditch `curl`. However, the patch likely means we should set minimum Ansible version to v2.4 or so.

File: vars/Ubuntu_vivid.yml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ---
 grsecurity_build_apt_packages:
+  - bc
```
Comment (conorsch) [inline vars/Ubuntu_vivid.yml]: :+1: Haven't been testing against these hosts, but `bc` definitely required.

File: tasks/main-stable.yml [hunk 3]
```diff
@@ -19,6 +19,7 @@
   # according to patch type 'test' or 'stable'.
   grsecurity_urls:
     patch_type: "{{ grsecurity_build_patch_type }}"
+  tags: always
```
Comment (conorsch) [inline tasks/main-stable.yml]: I believe this will break the `unofficial` scenario—testing locally...

Comment (conorsch) [reply tasks/main-stable.yml]: Nope, I'm wrong: `molecule test -s unofficial` gets to the build command without issue. Must be the dynamic import logic now, so tags for skipped includes aren't referenced (even via `always`).

File: tasks/compile.yml [hunk 4]
```diff
@@ -5,15 +5,19 @@
...
 - name: Build the grsecurity-patched kernel.
   command: "{{ grsecurity_build_compile_command }}"
   args:
     chdir: "{{ grsecurity_build_linux_source_directory }}"
-  environment:
-    CONCURRENCY_LEVEL: "{{ ansible_processor_vcpus }}"
-    # Conditionally enable ccache by prepending the ccache lib directory
-    # to PATH, which will override the system gcc and and g++ binaries.
-    PATH: "{% if grsecurity_build_use_ccache == true %}/usr/lib/ccache:{% endif %}{{ ansible_env.PATH }}"
+  environment: "{{ _grsec_build_env|combine(grsecurity_build_env_vars) }}"
```
[the remark this answers is not in the document]
Comment (msheiny) [reply tasks/compile.yml]: oooo yeah i like setting that in defaults instead - good idea buddy 👍

**Pull request merged**